mbed_tools_ci_scripts/create_news_file.py

```diff
@@ -0,0 +1,72 @@
+"""Easy news files generation.
+
+Usage:
+    create-news-file "Fixed a bug" --type bugfix
+"""
+import argparse
+import enum
+import pathlib
+from datetime import datetime
+
+from mbed_tools_ci_scripts.utils.configuration import configuration, ConfigurationVariable
+
+
+NEWS_DIR = configuration.get_value(ConfigurationVariable.NEWS_DIR)
+
+
+class NewsType(enum.Enum):
+    """Describes the type of news we're writing."""
+
+    BUGFIX = 0
+    DOC = 1
+    FEATURE = 2
+    MAJOR = 3
+    MISC = 4
+    REMOVAL = 5
+
+    @property
+    def extension(self) -> str:
+        """Returns lowercase extension for self. Example: "bugfix", "doc", etc."""
+        return self.name.lower()
+
+
+def create_news_file(news_text: str, news_type: NewsType) -> pathlib.Path:
+    """Facilitates creating a news file, determining it's file name based on the type."""
+    file_path = determine_news_file_path(news_type)
+    _write_file(file_path, news_text)
+    return file_path
+
+
+def determine_news_file_path(news_type: NewsType) -> pathlib.Path:
+    """Returns an available file path for given news type."""
+    news_file_name = _determine_todays_news_file_name()
+    news_file_path = pathlib.Path(NEWS_DIR, f"{news_file_name}.{news_type.extension}")
+    inc = 0
+    while news_file_path.exists():
+        inc += 1
+        news_file_path = news_file_path.with_name(f"{news_file_name}{inc:0=2}.{news_type.extension}")
+    return news_file_path
+
+
+def _write_file(file_path: pathlib.Path, text: str) -> None:
+    file_path.parent.mkdir(parents=True, exist_ok=True)
+    file_path.write_text(text)
+
+
+def _determine_todays_news_file_name() -> str:
+    return datetime.now().strftime("%Y%m%d")
+
+
+def main() -> None:
+    """Parses cli arguments and creates a news file."""
+    parser = argparse.ArgumentParser()
+    news_type_choices = {t.extension: t for t in NewsType}
+    parser.add_argument("news_text", help="Contents of the news file.")
```

Comment:
Should we validate the text is only a single line? `NewsFileValidator` checks for that specifically right?

Reply:
Not a bad idea, I'll see what can be done.

Reply:
Fixed https://github.com/ARMmbed/mbed-tools-ci-scripts/pull/34/commits/13f8ac9fb04de9c67a8b1f3264defb30c05617c5.